### PR TITLE
RCIS documentation

### PR DESCRIPTION
**Pull request by anindex**
This PR proposes RCIS documentations.

### CODE REVIEW

File: doc/RCIS.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# RoboComp Innermodel Simulator (RCIS)
+
+
+## Introduction
+`RCIS` is one of the core components of RoboComp, which creates a simulation environment for operational testing of individual components or system testing of a combined aggregated system of components. To be concrete, `RCIS` visualizes the [RoboComp's internal representation of the world and the robot itself](innermodel.md).
```
Comment (orensbruli) [inline doc/RCIS.md]:
```suggestion
`RCIS` is one of the core tools of RoboComp, which creates a simulation environment for operational testing of individual components or system testing of a combined aggregated system of components. To be concrete, `RCIS` simulate a [RoboComp's representation of the world and the robot itself](innermodel.md).
```

File: doc/RCIS.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+`RCIS` is implemented in C++ using **QGLWidget** library to draw various sprites such as planes, walls, boxes, robots, etc that represent the simulation world. The picture below describes a general view of implementation parts of `RCIS`:
```
Comment (orensbruli) [inline doc/RCIS.md]:
```suggestion
`RCIS` is implemented in C++ using **QGLWidget** library to draw various sprites such as planes, walls, boxes, robots, etc that represent the simulated world. The picture below describes a general view of implementation parts of `RCIS`:
```

File: doc/RCIS.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+RCIS consists of two main implementation parts: **Innermodel Manager** and **RoboComp interfaces: Sensors and Actuators**. The **Innermodel Manager** part deals with internal physics and transformation between simulated objects. Currently, these are the list of implemented RoboComp interfaces that are ready for simulation:
+- Camera
+- DifferentialRobot
+- IMU
+- JointMotor
+- Laser
+- RGBD
+- Display
+
+When starting simulation, `RCIS` will read an XML file for simulation world definitions, then it populates the objects defined in the XML file and start component servers for each of these interfaces. These component servers dedicatedly process information captured from or interacted with the simulation environment.
```
Comment (orensbruli) [inline doc/RCIS.md]:
```suggestion
When starting simulation, `RCIS` will read an XML file with the definitions of the simulated world and robot, then it populates the objects defined in the XML file and starts component servers for each of these interfaces. These component servers dedicatedly process information captured from or interacted with the simulation environment.
```

File: doc/RCIS.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+In this section, we will demonstrate an example usage of `RCIS`.
+### Run the RCIS
+Now let's run the simulator with "simple world" definitions `simpleworld.xml`
+```
+cd ~/robocomp/files/innermodel
+rcis simpleworld.xml
+```
+
+Now, RCIS should be up and running with a simple robot endowed with a laser and an RGBD camera, moving on a wooden floor. Please interact with the GUI window by dragging the mouse to see the world with different views.
```
Comment (orensbruli) [inline doc/RCIS.md]:
```suggestion
Now, RCIS should be up and running with a simple robot endowed with a differential base, a laser, and an RGBD camera. The robot will be shown in an environment with four walls, some boxes, and a wooden floor. Please interact with the GUI window by dragging the mouse to see the world with different views.
```

File: doc/RCIS.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+### Using the keyboard as a JoyStick
+
+To control the robot in the simulation world, we must first install the basic RoboComp components:
```
Comment (orensbruli) [inline doc/RCIS.md]:
```suggestion
To control the robot in the simulated world, we must first install the basic RoboComp components:
```

**Comment by orensbruli** [comment]
Maybe we need to add a note about the ports defined for the simulated components on the .xml and explain that those should match the ones that the other components need to connect to.
In the example, the differential robot is defined to be started on 10004 port on the simpleworld.xml and by default, the keyboardrobotcontroller/etc/config is configured to connect to the same port.
Line in the simpleworld.xml:
<differentialrobot id="base" port="10004">
Line in the keyboardrobotcontroller/etc/config:
DifferentialRobotProxy = differentialrobot:tcp -h localhost -p 10004

Thank you!

**Comment by anindex** [comment]
I have included the part you suggested as **Resolving Issues** section.

**Pull request merged**